ntoskrnl/mm/ARM3/mdlsup.c

```diff
@@ -1179,39 +1192,22 @@ MmProbeAndLockPages(IN PMDL Mdl,
         // Assume failure and check for non-mapped pages
         //
         *MdlPages = LIST_HEAD;
+
+        /* Check this is actually present in the NonPagedPool case */
         while (
 #if (_MI_PAGING_LEVELS == 4)
-               (PointerPxe->u.Hard.Valid == 0) ||
+            (PointerPxe->u.Hard.Valid == 0) ||
 #endif
 #if (_MI_PAGING_LEVELS >= 3)
-               (PointerPpe->u.Hard.Valid == 0) ||
+            (PointerPpe->u.Hard.Valid == 0) ||
 #endif
-               (PointerPde->u.Hard.Valid == 0) ||
-               (PointerPte->u.Hard.Valid == 0))
+            (PointerPde->u.Hard.Valid == 0) ||
+            (PointerPte->u.Hard.Valid == 0))
         {
-            //
-            // What kind of lock were we using?
-            //
-            if (UsePfnLock)
-            {
-                //
-                // Release PFN lock
-                //
-                MiReleasePfnLock(OldIrql);
-            }
-            else
-            {
-                /* Release process working set */
-                MiUnlockProcessWorkingSet(CurrentProcess, PsGetCurrentThread());
-            }
-
-            //
-            // Access the page
-            //
+            /* Access the page if this not non-paged pool */
```

Comment:
```suggestion
            /* Access the page if this is not non-paged pool */
```

Reply:
Comment fixed, I originally wrote it for a different code.